applications/StructuralMechanicsApplication/custom_elements/membrane_element.cpp

```diff
@@ -0,0 +1,1143 @@
+        // getting information for integration
+        const double integration_weight_i = r_integration_points[point_number].Weight();
+        const Matrix& shape_functions_gradients_i = r_shape_functions_gradients[point_number];
+
+        CovariantBaseVectors(current_covariant_base_vectors,shape_functions_gradients_i,"current");
+        CovariantBaseVectors(reference_covariant_base_vectors,shape_functions_gradients_i,"reference");
+
+        CovariantMetric(covariant_metric_current,current_covariant_base_vectors);
+        CovariantMetric(covariant_metric_reference,reference_covariant_base_vectors);
+        ContravariantMetric(contravariant_metric_reference,covariant_metric_reference);
+
+        ContraVariantBaseVectors(reference_contravariant_base_vectors,contravariant_metric_reference,reference_covariant_base_vectors);
+
+        TransformBaseVectors(transformed_base_vectors,reference_contravariant_base_vectors);
+
+        InPlaneTransformationMatrix(inplane_transformation_matrix_material,transformed_base_vectors,reference_contravariant_base_vectors);
+
+        JacobiDeterminante(detJ,reference_covariant_base_vectors);
+        StressPk2(stress,contravariant_metric_reference,covariant_metric_reference,covariant_metric_current,
+            transformed_base_vectors,inplane_transformation_matrix_material);
+
+        Matrix material_tangent_modulus = ZeroMatrix(dimension);
+        MaterialTangentModulus(material_tangent_modulus,contravariant_metric_reference,covariant_metric_reference,covariant_metric_current,
+            inplane_transformation_matrix_material);
+
+        for (SizeType dof_s=0;dof_s<number_dofs;++dof_s){
+            for (SizeType dof_r=0;dof_r<number_dofs;++dof_r){
+
+                //do not calculate symmetric entries
+                if(dof_s>dof_r){
+                    if (point_number==(r_integration_points.size()-1)) rStiffnessMatrix(dof_s,dof_r) = rStiffnessMatrix(dof_r,dof_s);
+                }
+                else{
+                    MaterialStiffnessMatrixEntryIJ(rStiffnessMatrix(dof_s,dof_r),
+                        material_tangent_modulus,dof_s,dof_r,shape_functions_gradients_i,
+                        current_covariant_base_vectors,inplane_transformation_matrix_material);
+                    InitialStressStiffnessMatrixEntryIJ(rStiffnessMatrix(dof_s,dof_r),
+                        stress,dof_s,dof_r,shape_functions_gradients_i,
+                        inplane_transformation_matrix_material);
+                    rStiffnessMatrix(dof_s,dof_r) *= detJ*integration_weight_i*thickness;
+                }
+            }
+        }
+    }
+}
+
+void MembraneElement::TransformBaseVectors(array_1d<Vector,2>& rBaseVectors,
+     const array_1d<Vector,2>& rLocalBaseVectors){
+
+    // create local cartesian coordinate system aligned to global material vectors (orthotropic)
+    if (Has(LOCAL_MATERIAL_AXIS_1) && Has(LOCAL_MATERIAL_AXIS_2)){
```

Comment:
do you always have to give both?
I thought we discussed to have them also separately?

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/2c70d79ac4261c955ada0e4058eb240a5038c9c5/applications/StructuralMechanicsApplication/custom_elements/membrane_element.cpp#L708